virtualizarr/manifests/array.py

```diff
@@ -79,12 +84,20 @@ def manifest(self) -> ChunkManifest:
         return self._manifest
 
     @property
-    def zarray(self) -> ZArray:
+    def zarray(self) -> ArrayMetadata:
         return self._zarray
 
     @property
     def chunks(self) -> tuple[int, ...]:
-        return tuple(self.zarray.chunks)
+        """
+        Individual chunk size by number of elements.
+        """
+        if isinstance(self._zarray.chunk_grid, RegularChunkGrid):
```

Comment:
this is a nice check, but I think we could actually perform this check at `ManifestArray` construction time, because right now a lot of other things will break if the chunk grid is not regular.